modules/apps/foundation/web-proxy/web-proxy-web/src/main/java/com/liferay/web/proxy/web/internal/servlet/filter/WebProxyServletFilter.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.web.proxy.web.internal.servlet.filter;
+
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.servlet.BaseFilter;
+import com.liferay.portal.kernel.servlet.HttpHeaders;
+import com.liferay.portal.kernel.util.ContentTypes;
+import com.liferay.web.proxy.web.internal.servlet.request.WebProxyServletRequest;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Daniel Couso
+ */
+@Component(
+	immediate = true,
+	property = {
+		"before-filter=Valid Host Name Filter", "dispatcher=FORWARD",
```

Comment:
Why is this filter associated with the external servlet container (of the portal) instead of being associated with the servlet context of the web proxy module?

Reply:
This filter has to be applied before the request inputStream object has been consumed and this can happen when any filter of the chain or any servlet calls getParameter and other methods.

As far as I can understand, using the "internal" servlet container would not guarantee this order making this fix approach useless.

Please, let me know whether my assumptions are correct or whether there would be other possibilities to configure this filter to achieve the same results.

Thanks in advance

Reply:
Have you tried?

Reply:
I think I did in my first attempts without success but I will check this again and let you know.

Reply:
I have done the tests without success.. If I configure the filter with:
"osgi.http.whiteboard.filter.dispatcher=FORWARD",
"osgi.http.whiteboard.filter.dispatcher=REQUEST",
"osgi.http.whiteboard.filter.servlet=org.portletbridge.portlet.PortletBridgeServlet",
"osgi.http.whiteboard.filter.name=com.liferay.web.proxy.web.internal.servlet.filter.WebProxyServletFilter",
"osgi.http.whiteboard.filter.pattern=/*",
"service.ranking:Integer=10000"

this filter is going to be called too late and the request's inputStream object is going to be already consumed by other external filters making it unavailable for the PortletBridgeServlet which needs to find it "readable".

Please, let me know if anything else needs to be checked or reviewed.

Reply:
sounds reasonable!